### DISCUSSION TITLE
Model cannot be found in Phalcon controller

### DISCUSSION THREAD

**Discussion by kinhdang** [phalcon/cphalcon#16389] [General]
Hi all, I'm new to Phalcon and I'm trying to do a simple thing, it run properly in windows, but when i move to server in centOS, it get  error 

Uncaught Error: Class 'test' not found in /var/www/html/qlhoso/app/controllers/TestController.php:10\nStack trace:\n#0 [internal function]: TestController->indexAction()\n#1 [internal function]: Phalcon\\Dispatcher\\AbstractDispatcher->callActionMethod(Object(TestController), 'indexAction', Array)\n#2 [internal function]: Phalcon\\Dispatcher\\AbstractDispatcher->dispatch()\n#3 /var/www/html/qlhoso/public/index.php(90): Phalcon\\Mvc\\Application->handle('/test')\n#4 {main}\n  thrown in /var/www/html/qlhoso/app/controllers/TestController.php on line 10

//////////////////////////
// Register an autoloader
$loader = new Loader();

$loader->registerDirs(
    [
        APP_PATH . '/controllers/',
        APP_PATH . '/models/',
    ]
);

$loader->register();

file structure:

![Untitled_file_ok](https://github.com/phalcon/cphalcon/assets/9839084/688bb9a5-07e1-4f33-a382-a0c67ed8973f)

php_info()

Phalcon

Author	Phalcon Team and contributors
Version	4.1.0
Build Date	Dec 31 2020 11:19:09
Powered by Zephir	Version 0.12.19-4345593

psr

Version	1.0.1
Released	2020-11-12
Authors	John Boehr <jbboehr@gmail.com> (lead)
PSR-3 Log Version	1.0.0
PSR-6 Cache Version	1.0.0
PSR-7 Http Message Version	1.0.0
PSR-11 Container Version	1.0.0
PSR-13 Link Version	1.0.0
PSR-14 Event Dispatcher	1.0.0
PSR-15 HTTP Handlers (Server Handler)	1.0.0
PSR-15 HTTP Handlers (Middleware)	1.0.0
PSR-16 Simple Cache Version	1.0.0
PSR-17 HTTP Factories	1.0.0
PSR-18 HTTP Client	1.0.0

**Comment by tuannguyen29** [comment]
- check your `class`, `namespace` names
- if controller name is `TestController` then class name is `TestController`
- on linux, phalcon is case sensitive
- check the route again

**Reply by kinhdang** [reply]
Thanks,

file controller "TestController.php"  and model "Test.php" below

<?php
use Phalcon\Mvc\Controller;

class TestController extends Controller
{
    public function indexAction()
    {     
        Phalcon\Tag::setTitle('..:: Login'); 
         
       $this->view->test =test::find();
    }
    
}

?>

<?php
use Phalcon\Mvc\Model;

class test extends Model
{
    public $id;
    public $name; 
}

?>

error when runing in centos.

 PHP Fatal error:  Uncaught Error: Class 'test' not found in /var/www/html/qlhoso/app/controllers/TestController.php:10\nStack trace:\n#0 [internal function]: TestController->indexAction()\n#1 [internal function]: Phalcon\\Dispatcher\\AbstractDispatcher->callActionMethod(Object(TestController), 'indexAction', Array)\n#2 [internal function]: Phalcon\\Dispatcher\\AbstractDispatcher->dispatch()\n#3 /var/www/html/qlhoso/public/index.php(90): Phalcon\\Mvc\\Application->handle('/test')\n#4 {main}\n  thrown in /var/www/html/qlhoso/app/controllers/TestController.php on line 10

**Reply by tuannguyen29** [reply]
change `test::find();` to `Test::find();`
and `use Model\Test` at top of TestController file.

**Reply by kinhdang** [reply]
![Untitled_file_ok](https://github.com/phalcon/cphalcon/assets/9839084/1ff536cd-1acd-4c7f-88ba-decdba656d40)

still something wrong, 😕

**Comment by hakimio** [comment]
Use Phalcon loader:
`loader.php`
```php
use Phalcon\Autoload\Loader;

$loader = new Loader();

$loader->setFiles([BASE_PATH.'/composer_vendor/autoload.php']);

/**
 * We're a registering a set of directories taken from the configuration file
 */
$loader->setDirectories([
    APP_PATH . '/controllers',
    APP_PATH . '/models'
]);

$loader->setNamespaces([
    'MyApp' =>  APP_PATH,
    'MyApp\Controllers' =>  APP_PATH . '/controllers',
    'MyApp\Models' =>  APP_PATH . '/models',
    'MyApp\Utils' => APP_PATH . '/utils'
]);

$loader->register();
```
Then import using `use MyApp\Models\Test;`

**Reply by kinhdang** [reply]
thanks,
